src/sync.rs

```diff
@@ -6,10 +6,13 @@
 //! [PEP 703]: https://peps.python.org/pep-703/
 use crate::{
     types::{any::PyAnyMethods, PyString, PyType},
-    Bound, Py, PyResult, PyVisit, Python,
+    Bound, Py, PyResult, Python,
 };
 use std::cell::UnsafeCell;
 
+#[cfg(not(Py_GIL_DISABLED))]
+use crate::PyVisit;
+
 /// Value with concurrent access protected by the GIL.
```

Comment:
I wonder if we should document that this type is unavailable on the freethreaded build and recommend use of atomics or `Mutex` as necessary. I guess in the future if we offer `PyMutex` as a type then we can recommend here too (as I understood it, if it blocks it will also release the GIL to help avoid deadlocking).

Reply:
I’ll take a look at adding and documenting safe wrappers for `PyMutex` that pyo3 can expose in a followup.